### PR TITLE
refactor: Updates project SDK and runtime

### PR DESCRIPTION
**Pull request by EspenAlbert**
Update project SDK and runtime for project resources.
It is necessary to have the latest SDK that includes the `tags` attribute on `admin.Group`

Link to any related issue(s):  CLOUDP-243393

This PR does `NOT` add the `tags` attribute, only:

1. Upgrade the SDK and cleanup the code in `resource.go`
2. Upgrade the provider runtime `go1.x`  - > `provided.al2`
3. Support creating test files when an organization API key already exists

`cfn test --function-name TestEntrypoint --verbose --region eu-south-2`

image

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Manual QA performed:

- [x] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

File: cfn-resources/project/cmd/resource/resource.go [hunk 1]
```diff
@@ -26,44 +26,41 @@ import (
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
-	"go.mongodb.org/atlas-sdk/v20231115002/admin"
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
 )
 
 var CreateRequiredFields = []string{constants.OrgID, constants.Name}
+var ReadRequiredFields = []string{constants.ID}
```
Comment (lantoli) [inline cfn-resources/project/cmd/resource/resource.go:33]: knit: have only one var ReadUpdateDeleteRequiredFields

File: cfn-resources/project/cmd/resource/resource.go [hunk 2]
```diff
@@ -86,12 +83,20 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
...
+	projectID := project.GetId()
 
 	// Add ApiKeys
 	if len(currentModel.ProjectApiKeys) > 0 {
 		for _, key := range currentModel.ProjectApiKeys {
-			_, res, err := atlasV2.ProgrammaticAPIKeysApi.UpdateApiKeyRoles(context.Background(), *project.Id, *key.Key, &admin.UpdateAtlasProjectApiKey{
-				Roles: key.RoleNames,
+			if key.Key == nil {
+				errorMessage := fmt.Sprintf("ApiKey is missing the configuration for projectID=%s", projectID)
+				_, _ = logger.Warn(errorMessage)
```
Comment (lantoli) [inline cfn-resources/project/cmd/resource/resource.go:93]: i would remove all logs that don't add anything, like this one. if we're already returning the error then no need to also log it.  i would search for "logger." and see if any one if needed at all

File: cfn-resources/project/cmd/resource/resource.go [hunk 3]
```diff
@@ -505,8 +468,9 @@ func getChangeInTeams(currentTeams []ProjectTeam, oTeams []admin.TeamRole) (newT
 func readTeams(teams []ProjectTeam) []admin.TeamRole {
 	var newTeams []admin.TeamRole
 	for _, t := range teams {
+		t := t
```
Comment (lantoli) [inline cfn-resources/project/cmd/resource/resource.go:471]: same

File: cfn-resources/project/cmd/resource/resource.go [hunk 4]
```diff
@@ -559,3 +523,30 @@ func getChangeInAPIKeys(currentKeys []ProjectApiKey, previousKeys []ProjectApiKe
...
+		}
+	}
+	// Add Teams
+	if len(newTeams) > 0 {
+		_, _, err = atlasV2.TeamsApi.AddAllTeamsToProject(context.Background(), projectID, &newTeams).Execute()
+		if err != nil {
+			return "Error while adding team to project", err
+		}
+	}
+	// Update Teams
```
Comment (lantoli) [inline cfn-resources/project/cmd/resource/resource.go:544]: knit: not sure if these comments are needed or the code is clear enough (or could be made more clear so we don't need comments)

File: cfn-resources/project/template.yml [hunk 5]
```diff
@@ -11,18 +11,17 @@ Resources:
...
-      Runtime: go1.x
+      Runtime: provided.al2
       CodeUri: bin/
 
   TestEntrypoint:
     Type: AWS::Serverless::Function
     Properties:
       Handler: bootstrap
-      Runtime: go1.x
+      Runtime: provided.al2
```
Comment (lantoli) [inline cfn-resources/project/template.yml:21]: how is it in other resources? would it make a sense a follow-up PR to to the change in bulk for all resources?

**Review by lantoli — APPROVED** [review]

File: cfn-resources/project/cmd/resource/resource.go (hunk 1 above)
Comment (EspenAlbert) [reply cfn-resources/project/cmd/resource/resource.go]: Solved in 1c80bca

File: cfn-resources/project/cmd/resource/resource.go (hunk 2 above)
Comment (EspenAlbert) [reply cfn-resources/project/cmd/resource/resource.go]: Removed all log statements

File: cfn-resources/project/cmd/resource/resource.go [hunk 6]
```diff
@@ -469,18 +431,19 @@ func readProjectSettings(atlasV2 *admin.APIClient, id string, currentModel *Mode
 func getChangeInTeams(currentTeams []ProjectTeam, oTeams []admin.TeamRole) (newTeams []admin.TeamRole,
 	changedTeams []admin.TeamRole, removeTeams []admin.TeamRole) {
 	for _, nTeam := range currentTeams {
+		nTeam := nTeam
```
[the remark this answers is not in the document]
Comment (EspenAlbert) [reply cfn-resources/project/cmd/resource/resource.go]: Thanks for the explanation 😁  updated in 1c80bca

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Pull request merged**